packages/rum-core/test/common/fetch-patch.spec.js

```diff
@@ -25,28 +25,31 @@
 
 import patchSubscription from './patch'
 import { globalState } from '../../src/common/patching/patch-utils'
+import { FETCH_SOURCE } from '../../src/common/constants'
 
 describe('fetchPatch', function() {
   var events = []
   var cancelFn
 
   beforeAll(function() {
     cancelFn = patchSubscription.subscribe(function(event, task) {
-      events.push({
-        event,
-        task
-      })
+      if (task.source === FETCH_SOURCE) {
+        events.push({
+          event,
+          task
+        })
+      }
```

Comment:
I think this is part of the tests, in other words we add all of the events to the array to make sure we only get the ones that are relevant to the test.

We should improve the tests to account for this.

Reply:
Since we use a global `events` variable to account for this, Our tests are getting flaky and breaking sometime. Check this 
```
LOG: Object{event: 'schedule', task: Object{source: 'XMLHttpRequest.send', state: 'invoke', type: 'macroTask', ignore: undefined, data: Object{target: ..., method: ..., sync: ..., url: ..., args: ..., aborted: ...}}},
LOG: Object{event: 'invoke', task: Object{source: 'fetch', state: 'invoke', type: 'macroTask', data: Object{target: ..., method: ..., sync: ..., url: ..., args: ..., aborted: ..., error: ...}}},
LOG: Object{event: 'invoke', task: Object{source: 'XMLHttpRequest.send', state: 'invoke', type: 'macroTask', ignore: undefined, data: Object{target: ..., method: ..., sync: ..., url: ..., args: ..., aborted: ...}}}
```

Reply:
There seems to be a bug in one of our tests in `fetch-patch.spec.js` that ends the test before the async operation is done